src/glum/_glm.py

```diff
@@ -1472,13 +1477,19 @@ def covariance_matrix(
             if clusters is not None:
                 n_groups = len(np.unique(clusters))
                 grouped_gradient = _group_sum(clusters, gradient)
-                inner_part = grouped_gradient.T @ grouped_gradient
+                if not isinstance(grouped_gradient, tm.SplitMatrix):
+                    inner_part = grouped_gradient.T @ grouped_gradient
+                else:
+                    inner_part = grouped_gradient.sandwich(np.ones_like(y))
```

Comment:
I think it would be best if we avoided making a separate computation for a SplitMatrix and for other types. Actually, I don't think `grouped_gradient` can be a SplitMatrix. We should check this.

Reply:
Yes, the gradient could be a SplitMatrix and after `_group_sum` this is an array, I have updated this here. For other calculations we will have to update `tabmat` to remove seperate computations for a SplitMatrix.